src/common/libflux/buffer.h

```diff
@@ -11,6 +11,10 @@
 #ifndef FLUX_BUFFER_H
 #define FLUX_BUFFER_H
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #include <stdbool.h>
```

Comment:
You also wrap standard headers here, would it be cleaner to just move the `extern "C"` after all `#include`s?